src/Nethermind/Nethermind.Evm/VirtualMachine.cs

```diff
@@ -415,8 +414,10 @@ public TransactionSubstate Run<TTracingActions>(EvmState state, IWorldState worl
 
                             // TODO: check if here we need to deduct gas from currentGas.GasAvailable or gasForCodeDeposit?
                             // Are they even different? - yes they are different - currentState.GasAvailable > gasAvailableForCodeDeposit
-                            long contractCreationCompleteGas = currentState.Env.Witness.AccessForContractCreated(callCodeOwner);
-                            if (gasAvailableForCodeDeposit >= contractCreationCompleteGas)
+                            long gasToBeUsed = gasAvailableForCodeDeposit;
+                            bool isEnough = currentState.Env.Witness.AccessForContractCreated(callCodeOwner, ref gasToBeUsed);
+                            var contractCreationCompleteGas = gasAvailableForCodeDeposit - gasToBeUsed;
```

Comment:
do we write specialized functions for handling or modify how this part is handled?